### PR TITLE
Expose uriScheme through vscode.env

### PR DESCRIPTION
**Pull request by rebornix**
Right now when extensions attempt to construct a callback url for themselves, they can only guess by checking the appName, but we should be smart enough to expose this information, for example `vscode`, `code-oss`, `vscode-exoloration`.

### CODE REVIEW

**Review by joaomoreno — APPROVED** [review]
Makes sense.

File: src/vs/vscode.proposed.d.ts [hunk 1]
```diff
@@ -625,6 +625,11 @@ declare module 'vscode' {
 		 * An [event](#Event) that fires when the log level has changed.
 		 */
 		export const onDidChangeLogLevel: Event<LogLevel>;
+
+		/**
+		 * The url protocol of the editor, like 'vscode', 'vscode-insiders'.
+		 */
+		export const urlProtocol: string;
```
Comment (jrieken) [inline src/vs/vscode.proposed.d.ts]: Not sure, the thing is called ur**i**-handler, now we talk about ur**l** protocol. What's correct? Why being inconsistent? Also protocol vs scheme?

Comment (jrieken) [reply src/vs/vscode.proposed.d.ts]: We should also update this https://github.com/Microsoft/vscode/blob/a3fc8198a7a274b580e8fbf7664e13a8bd8df647/src/vs/vscode.d.ts#L6643

**Review by jrieken — CHANGES_REQUESTED** [review]

File: src/vs/vscode.proposed.d.ts (hunk 1 above)
Comment (rebornix) [reply src/vs/vscode.proposed.d.ts]: We should call it `uriScheme` as `uriHandler` stands for the application.

**Comment by rebornix** [comment]
As we discussed on API syncup, we are going to use the name `uriScheme` for consistency, so I'm going to merge this PR. We can make tweaks if there are any other feedback.

**Pull request merged**